tools/test.py

```diff
@@ -1479,6 +1500,17 @@ def Main():
   def DoSkip(case):
     return SKIP in case.outcomes or SLOW in case.outcomes
   cases_to_run = [ c for c in all_cases if not DoSkip(c) ]
+  if options.run != None:
+    # Must ensure the list of tests is sorted before selecting, to avoid
+    # silent errors if this file is changed to list the tests in a way that
+    # can be different in different machines
+    cases_to_run.sort(cmp=lambda a, b: cmp(a.case.file, b.case.file))
+    cases_to_run.sort(cmp=lambda a, b: cmp(a.case.mode, b.case.mode))
+    cases_to_run.sort(cmp=lambda a, b: cmp(a.case.arch, b.case.arch))
```

Comment:
If the intention is to sort based on file first and if two files are same then based on mode and if modes are the same then based on arch, then we should use the `key` parameter, like this

    cases_to_run.sort(key=lambda c: (c.file, c.mode, c.arch))